### ISSUE TITLE
[API?] Should API parameters (both URL and POST) be explicitly URL-encoded?

### ISSUE THREAD

**Issue by martinrotter** [marienfressinaud/FreshRSS#4176]
**Describe the bug**
I am RSS Guard developer and I am regularly testing FreshRSS plugin (Greader API) with docker's `freshrss/freshrss` stuff. In case of FreshRSS, my plugin URL-encodes all parameters, for example for `streamContents` API method, `feed/123` parameter gets encoded to `feed%2F123` and resulting URL looks like this: `HOST/reader/api/0/stream/contents/feed%2F123?output=json&n=%2`.

Similarly in other methods. This perfectly works for majority of users.

One of my users, however, hosts FreshRSS via some classic "webhosting" service (php/mysql) and does not use docker. He somehow installs it manually and claims to use PHP 7.4. His instance seems to NOT work with my plugin and the problems is that his environment DOES NOT want URL parameters to be URL-encoded. It only works when URL parameters are UN-encoded like this: `HOST/reader/api/0/stream/contents/feed/123?output=json&n=%2`.

Any ideas what could be causing this?

**Comment by Alkarex** [comment]
Yes, see https://freshrss.github.io/FreshRSS/en/admins/10_ServerConfig.html
In Apache, `AllowEncodedSlashes On` is a requirement for using the Google Reader API
It would not be easy to allow something else without breaking the existing clients

**Comment by Alkarex** [comment]
By the way, let me know if we could discuss anything more regarding RSS Guard.
During the last check, which is already a bit old, the synchronisation was a bit slow, but maybe that could be updated :-)
https://github.com/FreshRSS/FreshRSS#apis--native-apps

**Comment by Frenzie** [comment]
@Alkarex Could we simply put that AllowEncodedSlashes directive [here](
https://github.com/FreshRSS/FreshRSS/blob/cf433d4d79fee49d5376ece9c50e62910239221f/p/api/.htaccess)? (Or in the proper equivalent if that's not the right location; just dropping this thought before I forget about it.) Or is that only for the VirtualHost definition.

**Comment by Alkarex** [comment]
@Frenzie Good suggestion, but I do not think it would be allowed in a `.htaccess` https://httpd.apache.org/docs/current/mod/core.html#allowencodedslashes
To be double-checked

**Comment by martinrotter** [comment]
> By the way, let me know if we could discuss anything more regarding RSS Guard. During the last check, which is already a bit old, the synchronisation was a bit slow, but maybe that could be updated :-) https://github.com/FreshRSS/FreshRSS#apis--native-apps

Well, could you please test with latest development build?

https://github.com/martinrotter/rssguard/releases/tag/devbuild

Sync speed should be relatively good. Also note that when setting up the account in rss guard, there are some options which influence speed of sync - one of them is what I call "Intelligent synchronization algorithm" checkbox.

Please, test and report to me. The synchronization logic got some fixes in last months and should be relatively fast - especially subsequent synces with that option enabled.

**Comment by martinrotter** [comment]
Sample video of speed - fetches 10 000 messages with Intelligent sync alg. enabled. Subsequent syncing is almost instant. And this is remote service. Locally selfhosted will be probably faster.

https://www.youtube.com/watch?v=H0HM-FDxV9Q

**Comment by martinrotter** [comment]
@Alkarex Do you need any more help from me with assessing the speed/quality of RSS Guard sync in order to obtain higher stars rating in your readme?
